### ISSUE TITLE
Clarification/documentation for CS9236

### ISSUE THREAD

**Issue by DanielRouxSA** [dotnet/roslyn#75166]
**Version Used**:

C# 12 on .NET 8

**Steps to Reproduce**:

Here is a small snippet of code lifted directly from one of my projects that triggers the CS9236 analyzer:

```CS
decimal standardSeasonalFee = currentContractPeriods
    .Sum((contract) => contract.product?.features
        .Sum((feature) => // This is where CS9236 is triggered.
        {
            return feature.standardSeasonalFee ?? 0M;
        }) ?? 0M
    );
```

Now, there may be ways I can write this code that would be more elegant/not raise CS9236, but that is not quite what I'm after at the moment.

I want to understand the reason this code triggers that analyzer, or rather, I want to understand the analyzer rule in the first place. However, there does not appear to be any documentation anywhere about it. I have tried to do what the message suggests - explicitly declaring the parameter types (on both the inner and outer lambda) as well as explicitly providing the generic type arguments (also on both inner and outer Sum calls), but the analyzer is still triggered.

**Diagnostic Id**:

CS9236	Compiling requires binding the lambda expression at least 100 times. Consider declaring the lambda expression with explicit parameter types, or if the containing method call is generic, consider using explicit type arguments.

**Comment by huoyaoyuan** [comment]
See #74799

**Comment by DanielRouxSA** [comment]
> See #74799

Ah thank you. I'm not sure why I did not find that while searching the repository prior to opening this issue.
